pkg/app/piped/cloudprovider/lambda/client.go

```diff
@@ -108,56 +92,34 @@ func (c *client) IsFunctionExist(ctx context.Context, name string) (bool, error)
 	input := &lambda.GetFunctionInput{
 		FunctionName: aws.String(name),
 	}
-	_, err := c.client.GetFunctionWithContext(ctx, input)
+	_, err := c.client.GetFunction(ctx, input)
 	if err != nil {
-		if aerr, ok := err.(awserr.Error); ok {
-			switch aerr.Code() {
-			case lambda.ErrCodeInvalidParameterValueException:
-				return false, fmt.Errorf("invalid parameter given: %w", err)
-			case lambda.ErrCodeServiceException:
-				return false, fmt.Errorf("aws lambda service encountered an internal error: %w", err)
-			case lambda.ErrCodeTooManyRequestsException:
-				return false, fmt.Errorf("request throughput limit was exceeded: %w", err)
+		var nfe *types.ResourceNotFoundException
+		if errors.As(err, &nfe) {
```

Comment:
Because we don't need to use `nfe` after this so we can use `Is` instead of `As` to check the type.

Reply:
Oh, my bad. That `ResourceNotFoundException` is a struct, not a variable.
Your check is fine.

Reply:
In that case, `errors.Is(err, &types.ResourceNotFoundException{})` would not be caught 😢 
the error returns from API is not a dummy object, it's properties so that `errors.Is()` would return false.